### PR TITLE
Refactor deploy command

### PR DESCRIPTION
**Pull request by msteffen**
so that various deploy targets are in separate sub-commands.

- Regenerated the docs for the new command
- Updated our getting started guide to use `pachctl deploy basic`

Testing:
- I only tested this manually. I generated a kubernetes manifest using the old pachctl deploy command, and again using the new one, and looked for diffs between them (using meld).

This test will be hard to automate for the time being because the codec library (that we use to generate k8s manifests) doesn't seem to consistently canonicalize maps. Top level objects and maps are canonicalized, but some number of nested maps are not. I believe this is a bug in codec, and I hope to be able to reproduce it for them so that we can start writing tests for this code, but for now I confirmed that maps were the same visually

### CODE REVIEW

**Comment by msteffen** [comment]
@jdoliner whenever you get a chance

File: src/server/pkg/deploy/cmds/cmds.go [hunk 1]
```diff
@@ -17,92 +17,115 @@ import (
 	"go.pedge.io/pkg/exec"
 )
 
+func maybeKcCreate(dryRun bool, manifest *bytes.Buffer) error {
+	if dryRun {
+		n, err := os.Stdout.Write(manifest.Bytes())
+		if err == nil && n < manifest.Len() {
+			return io.ErrShortWrite
+		}
+		return err
```
Comment (jdoliner) [inline src/server/pkg/deploy/cmds/cmds.go]: Nice thorough error checking here but you I believe you can get the same effect with:

```
_, err := os.Stdout.Write(manifest.Bytes())
return err
```

Go specifies in the semantics of `io.Writer` that:

>  Write must return a non-nil error if it returns n < len(p).

https://golang.org/pkg/io/#Writer

Comment (msteffen) [inline src/server/pkg/deploy/cmds/cmds.go]: Oh, ez. Fixed.

**Comment by msteffen** [comment]
Deleting branch because CI was broken, so I couldn't submit this, and then the merge got too complicated (maybe I should've been retrying CI more aggressively—I'm not sure how so many PRs got submitted while CI was crashing, but in any case it seems to be healthier now that the time limit is extended)

**Pull request closed without merging**